cmd/lotus-storage-miner/proving.go

```diff
@@ -49,54 +51,41 @@ var provingFaultsCmd = &cli.Command{
 
 		ctx := lcli.ReqContext(cctx)
 
+		stor := store.ActorStore(ctx, apibstore.NewAPIBlockstore(api))
+
 		maddr, err := getActorAddress(ctx, nodeApi, cctx.String("actor"))
 		if err != nil {
 			return err
 		}
 
-		var mas miner.State
-		{
-			mact, err := api.StateGetActor(ctx, maddr, types.EmptyTSK)
-			if err != nil {
-				return err
-			}
-			rmas, err := api.ChainReadObj(ctx, mact.Head)
-			if err != nil {
-				return err
-			}
-			if err := mas.UnmarshalCBOR(bytes.NewReader(rmas)); err != nil {
-				return err
-			}
-		}
-
-		fmt.Printf("Miner: %s\n", color.BlueString("%s", maddr))
-
-		head, err := api.ChainHead(ctx)
+		mact, err := api.StateGetActor(ctx, maddr, types.EmptyTSK)
```

Comment:
This will end up loading blocks over the API to access the state. We could _also_ just add methods to the API to support specific queries, where necessary (e.g. "list all partitions", "count x, y, z", etc.).

I did it this way because it's simpler and we can always change our minds later if it becomes a performance issue.

Note: Replicating the "give me all the partitions" logic was a bit tricky because the structure of these objects depended on the chain height. It was doable, just not really worth it.